go/vt/automation/resharding_task.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package automation
+
+import (
+	"fmt"
+	"strings"
+
+	pb "github.com/youtube/vitess/go/vt/proto/automation"
+)
+
+// ReshardingTask is a cluster operation which allows to increase the number of shards.
+type ReshardingTask struct {
+}
+
+func splitShardListIntoKeyspaceAndShards(shardsWithKeyspace []string) (string, []string, error) {
+	var keyspace string
+	shards := make([]string, 0, len(shardsWithKeyspace))
+
+	for i, shard := range shardsWithKeyspace {
+		keyspaceAndShard := strings.Split(shard, "/")
+		if len(keyspaceAndShard) != 2 {
+			return "", nil, fmt.Errorf("Invalid shard: %v Required form: keyspace/shard-nr", shard)
+		}
+		if i == 0 {
+			keyspace = keyspaceAndShard[0]
+		} else {
+			if keyspace != keyspaceAndShard[0] {
+				return "", nil, fmt.Errorf("All shards must have the same keyspace. First seen keyspace: %v Wrong shard: %v", keyspace, shard)
+			}
+		}
+		shards = append(shards, keyspaceAndShard[1])
+	}
+
+	return keyspace, shards, nil
+}
+
+func selectAnyTabletFromShardByType(shard string, tabletType string) string {
+	return ""
+}
+
+func (t *ReshardingTask) run(parameters map[string]string) ([]*pb.TaskContainer, string, error) {
+	sourceShardsWithKeyspace := strings.Split(parameters["source_shard_list"], ",")
+	sourceRdonlyTablets := strings.Split(parameters["source_shard_rdonly_list"], ",")
+
+	keyspace, sourceShards, err := splitShardListIntoKeyspaceAndShards(sourceShardsWithKeyspace)
+	if err != nil {
+		return nil, "", err
+	}
+	destShardsWithKeyspace := strings.Split(parameters["dest_shard_list"], ",")
+	keyspaceDest, destShards, err := splitShardListIntoKeyspaceAndShards(destShardsWithKeyspace)
+	if keyspace != keyspaceDest {
+		return nil, "", fmt.Errorf("Source and Destination keyspace are not equal. Source: %v Dest: %v", keyspace, keyspaceDest)
+	}
+
+	var newTasks []*pb.TaskContainer
+	reparentTasks := NewTaskContainer()
+	for _, destShard := range destShards {
+		newMaster := selectAnyTabletFromShardByType(destShard, "master")
+		AddTask(reparentTasks, "ForceReparent", map[string]string{
```

Comment:
There is no ForceReparent task right now - I imagine that's intentional? Doesn't that cause the cluster operation to fail?